inst/include/Rcpp/String.h

```diff
@@ -64,10 +63,10 @@ namespace Rcpp {
             RCPP_STRING_DEBUG("String(const String&)");
         }
 
-        String(const String& other, const std::string& enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
+        String(const String& other, cetype_t enc) : data(other.get_sexp()), valid(true), buffer_ready(false) {
```

Comment:
This method feels a little weird:

1. It accepts a `String` object, which should have its own encoding; but
2. It _overrides_ that encoding with the encoding passed in using `set_encoding`.

I'm curious where / how this is used; it feels a little strange.